src/android/UploadTask.java

```diff
@@ -240,10 +294,24 @@ public Result doWork() {
         currentCall = httpClient.newCall(request);
 
         // Block until call is finished (or cancelled)
-        Response response;
+        Response response = null;
         try {
             if (!DEBUG_SKIP_UPLOAD) {
-                response = currentCall.execute();
+                try {
+                    try {
+                        concurrentUploads.acquire();
+                        try {
+                            response = currentCall.execute();
+                        } finally {
+                            concurrentUploads.release();
+                        }
+                    } catch (InterruptedException e) {
+                        e.printStackTrace();
```

Comment:
@zafirskthelifehacker here I think we might have an issue we are not telling the plug-in to do a retry if it's has been interrupted

Reply:
@ombr Yeah there should be a retry when it has been interrupted. I have added return Result.retry() for this catch statement.